### PR TITLE
doc(theme): Apply community style and reorganize accordingly

### PR DESCRIPTION
**Pull request by bobylito**
This layout based on the community standard as seen in the places and
helper websites. There are some missing elements:
 - proper writing of the doc
 - logo
 - style and images for example page
 - content for landing page

This is a first step :)

screenshot 2016-09-21 19 03 08

### CODE REVIEW

**Comment by vvo** [comment]
This look rad eheh!!!

In the screenshot you put, we miss the context of being in the "Guide" section, right now it looks like we are in the Styling section and that Integrating with other libraries is a sub category of Styling.

This is the feeling I got because of the big hero only displaying "Styling" then I linked the left nav to styling by default mentally.

Is having a more consistent navigation hints a second step maybe? Once we write more of it.

**Comment by bobylito** [comment]
Yep true, will change that!

**Comment by bobylito** [comment]
So now based on the nav group, it will highlight the current category. The build still fails but I can't reproduce the problem locally for the moment :(

**Comment by vvo** [comment]
> The build still fails but I can't reproduce the problem locally for the moment :(

I dont know metalsmith-sass but it seems to force the node-sass dependency, instead of making it a peerDependency (allowing you to upgrade easily).

Maybe you need to do `npm rebuild node-sass` on travis, but this might not be feasible given metalsmith-sass has it as a direct dependency.

File: docgen/middlewares.js [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
+import sass from 'metalsmith-sass';
 
-import inlineProps from './plugins/inlineProps/index.js';
-import renderer from './mdRenderer.js';
-import source from './plugins/source.js';
-import onlyChanged from './plugins/onlyChanged.js';
 import assets from './plugins/assets.js';
-import ignore from './plugins/ignore.js';
 import helpers from './plugins/helpers.js';
+import ignore from './plugins/ignore.js';
```
Comment (vvo) [inline docgen/middlewares.js]: We can enforce an eslint rule if you prefer alphabeticall orer of local imports. WFM :)

File: docgen/middlewares.js [hunk 2]
```diff
@@ -1,22 +1,25 @@
-import layouts from 'metalsmith-layouts';
 import headings from 'metalsmith-headings';
-import navigation from 'metalsmith-navigation';
+import layouts from 'metalsmith-layouts';
 import markdown from 'metalsmith-markdown';
+import msWebpack from 'ms-webpack';
+import navigation from 'metalsmith-navigation';
```
Comment (vvo) [inline docgen/middlewares.js]: Woah we kept that module, I thought it was really hard to work with it. But if it works now, wfm

File: docgen/plugins/postcss.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import postcss from 'postcss';
```
Comment (vvo) [inline docgen/plugins/postcss.js]: Do we still use postCSS in the end?

File: package.json [hunk 4]
```diff
@@ -45,15 +45,18 @@
     "metalsmith-layouts": "1.6.5",
     "metalsmith-markdown": "0.2.1",
     "metalsmith-navigation": "0.2.9",
+    "metalsmith-sass": "1.3.0",
     "ms-webpack": "2.0.0-alpha.1",
+    "postcss": "5.2.0",
```
Comment (vvo) [inline package.json]: Do we still need this?

File: docgen/src/images/concepts/conjunctive facets.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...font-weight:700;font-size:24px;fill:#000;">TV</text></g><g transform="matrix(1,0,0,1,-233.017,-1425.26)"><text x="832.333px" y="1883.43px" style="font-family:'Raleway-Regular', 'Raleway', sans-serif;font-size:24px;fill:#000;">Cat<tspan x="869.509px " y="1883.43px ">e</tspan>gory:</text><text x="944.773px" y="1883.43px" style="font-family:'Raleway-Bold', 'Raleway', sans-serif;font-weight:700;font-size:24px;fill:#000;">TV</text></g><g transform="matrix(1,0,0,1,-233.017,-1280.33)"><text x="832.333px" y="1883.43px" style="font-family:'Raleway-Regular', 'Raleway', sans-serif;font-size:24px;fill:#000;">Cat<tspan x="869.509px " y="1883.43px ">e</tspan>gory:</text><text x="944.773px" y="1883.43px" style="font-family:'Raleway-Bold', 'Raleway', sans-serif;font-weight:700;font-size:24px;fill:#000;">TV</text></g><g transform="matrix(1,0,0,1,-233.017,-1135.39)"><text x="832.333px" y="1883.43px" style="font-family:'Raleway-Regular', 'Raleway', sans-serif;font-size:24px;fill:#000;">Cat<tspan x="869.509px " y="1883.43px ">e</tspan>gory:</text><text x="944.773px" y="1883.43px" style="font-family:'Raleway-Bold', 'Raleway', sans-serif;font-weight:700;font-size:24px;fill:#000;">TV</text></g></g></svg>
```
Comment (vvo) [inline docgen/src/images/concepts/conjunctive facets.svg]: Can we remove those unused files or are they mandatory?

**Comment by bobylito** [comment]
thanks for the review @vvo

File: docgen/middlewares.js (hunk 2 above)
Comment (bobylito) [inline docgen/middlewares.js]: I had no problem using it. It generates a group of pages per nav group, WFM

File: docgen/middlewares.js (hunk 1 above)
Comment (bobylito) [inline docgen/middlewares.js]: Well it's more complex than just order by name, they are grouped and ordered. Is there a rule for that?

File: docgen/plugins/postcss.js (hunk 3 above)
Comment (bobylito) [inline docgen/plugins/postcss.js]: nope we can remove that. I will extract the metalsmith plugin as well.

File: docgen/middlewares.js (hunk 1 above)
Comment (vvo) [inline docgen/middlewares.js]: eslint-plugin-import is pretty advanced (we already use some rules), and maybe they have automatic fixable rules already :D https://github.com/benmosher/eslint-plugin-import

For now we are good, was just exploring the PR

Comment (bobylito) [inline docgen/middlewares.js]: Ok it's possible :D We'll see late :)

**Pull request merged**